mypy/build.py

```diff
@@ -544,12 +544,20 @@ def log(self, message: str) -> None:
 def remove_cwd_prefix_from_path(p: str) -> str:
     """Remove current working directory prefix from p, if present.
 
+    Also crawl up until a directory without __init__.py is found.
```

Comment:
This can definitely be refactored. In fact I think a lot of the main and build code should be refactored. But it doesn't feel like a priority -- since this PR it works well enough.